packages/stargate/src/aminotypes.ts

```diff
@@ -539,6 +548,12 @@ export class AminoTypes {
   }
 
   public toAmino({ typeUrl, value }: EncodeObject): AminoMsg {
+    if (this.unsupportedTypes.includes(typeUrl)) {
+      throw new Error(
+        `The message type '${typeUrl}' cannot be signed using the Amino JSON sign mode because this is not implemented on-chain.`,
+      );
+    }
+
     const converter = this.register[typeUrl];
     if (!converter) {
```

Comment:
This would become

```suggestion
    const converter = this.register[typeUrl];
    if (converter === "not_implemented_on_chain") {
      // error message from above
    } else if (!converter) {
```